### PR TITLE
Call the onload_callback in the "password" and "change password" modules

### LINKED ISSUE #2108: Password modules do not respect onload_callback of tl_member
**Affected version(s)**

Contao 4.4, 4.9, 4.10

**Description**

Both `ModuleRegistration` and `ModulePersonalData` check for onload_callbacks in tl_member since they use fields from the DCA. `ModulePassword` and `ModuleChangePassword` also loads `username` and `email` from the DCA but do not respect any onload_callbacks which imho is an inconsistency.

https://github.com/contao/contao/blob/a4dffd5495cace44948096d0e9481a32617b4d21/core-bundle/src/Resources/contao/modules/ModuleRegistration.php#L73-L88

### PR DESCRIPTION
**Pull request by leofeyer**
| Q                | A
| -----------------| ---
| Fixed issues     | Fixes #2108
| Docs PR or issue | https://github.com/contao/docs/pull/595

@bennyborn Please confirm that the changes work for you.

### CODE REVIEW

File: core-bundle/src/Resources/contao/modules/ModuleChangePassword.php [hunk 1]
```diff
@@ -70,6 +70,23 @@ protected function compile()
...
 
+		// Call onload_callback (e.g. to check permissions)
+		if (\is_array($GLOBALS['TL_DCA']['tl_member']['config']['onload_callback']))
+		{
+			foreach ($GLOBALS['TL_DCA']['tl_member']['config']['onload_callback'] as $callback)
+			{
+				if (\is_array($callback))
+				{
+					$this->import($callback[0]);
+					$this->{$callback[0]}->{$callback[1]}();
```
Comment (bytehead) [inline core-bundle/src/Resources/contao/modules/ModuleChangePassword.php:81]: Isn't an argument missing here (data container)?

File: core-bundle/src/Resources/contao/modules/ModulePassword.php [hunk 2]
```diff
@@ -63,6 +63,23 @@ protected function compile()
...
 
+		// Call onload_callback (e.g. to check permissions)
+		if (\is_array($GLOBALS['TL_DCA']['tl_member']['config']['onload_callback']))
+		{
+			foreach ($GLOBALS['TL_DCA']['tl_member']['config']['onload_callback'] as $callback)
+			{
+				if (\is_array($callback))
+				{
+					$this->import($callback[0]);
+					$this->{$callback[0]}->{$callback[1]}();
```
Comment (bytehead) [inline core-bundle/src/Resources/contao/modules/ModulePassword.php:74]: Same here.

File: core-bundle/src/Resources/contao/modules/ModuleChangePassword.php (hunk 1 above)
Comment (bytehead) [reply core-bundle/src/Resources/contao/modules/ModuleChangePassword.php:81]: Nvm: https://docs.contao.org/dev/reference/dca/callbacks/#config-onload
>Front end modules “Personal data” & “Registration”
No parameters.

File: core-bundle/src/Resources/contao/modules/ModulePassword.php (hunk 2 above)
Comment (bytehead) [reply core-bundle/src/Resources/contao/modules/ModulePassword.php:74]: Nvm: https://docs.contao.org/dev/reference/dca/callbacks/#config-onload
>Front end modules “Personal data” & “Registration”
No parameters.

**Review by bytehead — APPROVED** [review]

**Comment by bennyborn** [comment]
@leofeyer Looks good to me.

**Pull request merged**